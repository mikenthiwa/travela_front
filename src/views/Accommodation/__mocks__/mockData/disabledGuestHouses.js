const disabledGuestHouses = [
  {
    id: 'juhiuhoji0',
    houseName: 'Guest House A',
    location: 'Nairobi, Kenya',
    bathRooms: 5,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'bvuh9iei8bh',
        roomName: 'ZAPTIU',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'NBiojnn98u',
        roomName: 'AFEIK',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'jbg8oijjolp',
    houseName: 'Guest House B',
    location: 'Nairobi, Kenya',
    bathRooms: 3,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'bnvjhijkpj',
        roomName: 'APLI',
        roomType: 'Single',
        bedCount: 1,
        faulty: false,
      },
      {
        id: 'bnvuiohj',
        roomName: 'Onij',
        roomType: 'Single',
        bedCount: 4,
        faulty: false,
      }
    ]
  },
  {
    id: 'nbuiojkpl',
    houseName: 'Guest House C',
    bathRooms: 4,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Lagos, Nigeria',
    rooms: [
      {
        id: 'bvuijkok',
        roomName: 'Two',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
    ]
  },
  {
    id: 'nmbkihjokko',
    houseName: 'Guest House D',
    bathRooms: 1,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Kigali, Rwanda',
    rooms: [
      {
        id: 'jbihjjkpk',
        roomName: 'Roof',
        roomType: 'Suite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
];

export default disabledGuestHouses;
