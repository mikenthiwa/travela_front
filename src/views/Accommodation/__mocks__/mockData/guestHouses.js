const guestHouses = [
  {
    id: 'wrtiuy56_u',
    houseName: 'Guest House 1',
    location: 'Nairobi, Kenya',
    bathRooms: 5,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'tryuompl0',
        roomName: 'Nyati',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'trytrimo',
        roomName: 'Maputi',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'wrtnytom_9',
    houseName: 'Guest House 2',
    location: 'Nairobi, Kenya',
    bathRooms: 3,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'plmyt0pol',
        roomName: 'Santana',
        roomType: 'Single',
        bedCount: 1,
        faulty: false,
      },
      {
        id: 'plmmnuyt',
        roomName: 'Shiloh',
        roomType: 'Single',
        bedCount: 4,
        faulty: false,
      }
    ]
  },
  {
    id: 'bytrwqjk9',
    houseName: 'Guest House 3',
    bathRooms: 4,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Lagos, Nigeria',
    rooms: [
      {
        id: 'btrfdseq23',
        roomName: 'Aben',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'btrfdsnytr3',
        roomName: 'Aben',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'requntm0',
    houseName: 'Guest House 4',
    bathRooms: 1,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Kigali, Rwanda',
    rooms: [
      {
        id: 'mnytbvfdc9',
        roomName: 'Berger',
        roomType: 'Suite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
];

export default guestHouses;
