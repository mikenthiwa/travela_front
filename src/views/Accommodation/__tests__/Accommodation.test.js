import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import ConnectedAccommodation, { Accommodation } from '..';
import guestHouses from '../__mocks__/mockData/guestHouses';
import disabledGuestHouses from '../__mocks__/mockData/disabledGuestHouses';

const props = {
  guestHouses,
  disabledGuestHouses: disabledGuestHouses,
  fetchAccommodation: sinon.spy(),
  fetchDisabledAccommodation: sinon.spy(),
  restoreDisabledAccommodation: sinon.spy(),
  initFetchTimelineData: sinon.spy(),
  isLoading: false,
  createAccommodation: jest.fn(),
  createAccommodationLoading: false,
  editAccommodation: jest.fn(),
  shouldOpen: false,
  modalType: '',
  openModal: sinon.spy(),
  onNotificationToggle: jest.fn(),
  restoreGuestHouse: sinon.spy(),
  closeModal: sinon.spy(() => Promise.resolve()),
  getCurrentUserRole: ['Requester'],
  history: {
    push: jest.fn()
  },
  handleOnRestore: jest.fn(),
};

const initialState = {
  postAccommodationData: [],
  errors: [],
  modal: {
    shouldOpen: false,
    modalType: null
  }
};

const mockStore = configureStore();
const store = mockStore(initialState);

describe('<Accommodation />', () => {
  let wrapper;

  process.env.REACT_APP_CITY = 'https://maps.googleapis.com/maps/api/js?key=AIzaSyD-fvLImnNbTfYV3Pd1nJuK7NbzZJNr4ug&libraries=places';

  beforeEach(() => {
    wrapper = shallow(<Accommodation {...props} />);
  });

  it('should render the Accommodation page without crashing', () => {
    expect(wrapper.length).toBe(1);
    wrapper.unmount();
  });


  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it(`calls the fetchAccommodation method
      on componentDidMount`, () => {
    const { fetchAccommodation } = props;
    expect(fetchAccommodation.called).toEqual(true);
  });

  it(`calls the fetchDisabledAccommodation method
      on componentDidMount`, () => {
    const { fetchDisabledAccommodation } = props;
    expect(fetchDisabledAccommodation.called).toEqual(true);
  });

  it('should call restoreGuestHouse when the button is clicked', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'restoreGuestHouse');
    const button = wrapper.find('#restoreGuestHouseId');
    button.simulate('click');
    expect(instance.restoreGuestHouse).toBeCalled;
  });

  it('should call handleOnRestore', () => {
    const instance = wrapper.instance();
    jest.spyOn(instance, 'handleOnRestore');
    const button = wrapper.find('#handleOnRestoreId');
    button.simulate('click');
    expect(instance.handleOnRestore).toBeCalled;
  });

  it('renders the right number of centres', () => {
    const CentreGridWrapper = wrapper.find('WithLoading').dive()
      .find('CentreGrid');
    expect(CentreGridWrapper.dive().find('CentreCard').length).toBe(8);
  });

  xit('should redirect the user when the  user is not a super admin admin  or travel adminon componentDidMount method', () => {
    const { history } = props;
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Accommodation {...{ ...props, getCurrentUserRole: ' Administrator' }} />
        </MemoryRouter>
      </Provider>
    );
    expect(history.push).toHaveBeenCalledWith('/');
    wrapper.unmount();
  });

  it('should set `visibility` prop to `invisible` when cancel button is clicked on new accomodation modal', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Accommodation
            {...{ ...props, shouldOpen: true, modalType: 'new model' }}
          />
        </MemoryRouter>
      </Provider>
    );
    wrapper.find('button#cancel').simulate('click');
    expect(props.closeModal.calledWith(true, 'add accommodation')).toBeTruthy();
  });

  it('should call open modal with true when .restore-acc-btn is clicked', () => {
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <Accommodation
            {...{ ...props, shouldOpen: true, modalType: 'new model' }}
          />
        </MemoryRouter>
      </Provider>
    );
    wrapper.find('.restore-acc-btn').at(0).simulate('click');
    expect(props.openModal.calledWith(true, 'restore guesthouse')).toBeTruthy();
  });


});
