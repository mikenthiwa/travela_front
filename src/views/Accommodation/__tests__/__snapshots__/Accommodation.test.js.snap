// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accommodation /> renders correctly 1`] = `
<Fragment>
  <div
    className="rp-role__header"
  >
    <AccommodationPanelHeader
      openModal={[Function]}
    />
  </div>
  <div
    className="guesthouse-modal"
  >
    <Modal
      closeDeleteModal={null}
      closeModal={[Function]}
      customModalStyles=""
      customOverlayStyle=""
      modalBar={<div />}
      modalContentId=""
      modalId=""
      requestId=""
      showOverlay={true}
      title="Add Guest House"
      visibility="invisible"
      width="800px"
    >
      <NewAccommodation
        closeModal={[Function]}
        createAccommodation={[Function]}
        createAccommodationLoading={false}
        editAccommodation={[Function]}
        fetchAccommodation={[Function]}
        guestHouse={Object {}}
        initFetchTimelineData={[Function]}
        isSaving={false}
        modalType=""
        savingAccommodation={[Function]}
      />
    </Modal>
  </div>
  <Modal
    closeDeleteModal={null}
    closeModal={[Function]}
    customModalStyles="delete-checklist-item restore-model-content"
    customOverlayStyle=""
    modalBar={<div />}
    modalContentId=""
    modalId=""
    requestId=""
    showOverlay={true}
    title="Restore undefined"
    visibility="invisible"
    width=""
  >
    <span
      className="delete-checklist-item__disclaimer restore-checklist-items_span"
    >
      Are you sure you want to restore
      <strong />
    </span>
    <div
      className="delete-checklist-item__hr delete-checklist-item__left"
    />
    <div
      className="delete-checklist-item__footer delete-checklist-item__right"
    >
      <button
        className="delete-checklist-item__footer--cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
      <button
        className="bg-btn bg-btn--active restore-guestHouse-button"
        disabled={false}
        id="restoreGuestHouseId"
        onClick={[Function]}
        type="button"
      >
        <ButtonLoadingIcon
          buttonText="Restore"
          isLoading={false}
        />
      </button>
    </div>
  </Modal>
  <div
    className="table__container accommodation__container"
  >
    <WithLoading
      disabledGuestHouses={
        Array [
          Object {
            "bathRooms": 5,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House A",
            "id": "juhiuhoji0",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "bvuh9iei8bh",
                "roomName": "ZAPTIU",
                "roomType": "Ensuite",
              },
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "NBiojnn98u",
                "roomName": "AFEIK",
                "roomType": "Ensuite",
              },
            ],
          },
          Object {
            "bathRooms": 3,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House B",
            "id": "jbg8oijjolp",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 1,
                "faulty": false,
                "id": "bnvjhijkpj",
                "roomName": "APLI",
                "roomType": "Single",
              },
              Object {
                "bedCount": 4,
                "faulty": false,
                "id": "bnvuiohj",
                "roomName": "Onij",
                "roomType": "Single",
              },
            ],
          },
          Object {
            "bathRooms": 4,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House C",
            "id": "nbuiojkpl",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Lagos, Nigeria",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "bvuijkok",
                "roomName": "Two",
                "roomType": "Ensuite",
              },
            ],
          },
          Object {
            "bathRooms": 1,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House D",
            "id": "nmbkihjokko",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Kigali, Rwanda",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "jbihjjkpk",
                "roomName": "Roof",
                "roomType": "Suite",
              },
            ],
          },
        ]
      }
      error=""
      guestHouses={
        Array [
          Object {
            "bathRooms": 5,
            "genderPolicy": "unisex",
            "houseName": "Guest House 1",
            "id": "wrtiuy56_u",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "tryuompl0",
                "roomName": "Nyati",
                "roomType": "Ensuite",
              },
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "trytrimo",
                "roomName": "Maputi",
                "roomType": "Ensuite",
              },
            ],
          },
          Object {
            "bathRooms": 3,
            "genderPolicy": "unisex",
            "houseName": "Guest House 2",
            "id": "wrtnytom_9",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 1,
                "faulty": false,
                "id": "plmyt0pol",
                "roomName": "Santana",
                "roomType": "Single",
              },
              Object {
                "bedCount": 4,
                "faulty": false,
                "id": "plmmnuyt",
                "roomName": "Shiloh",
                "roomType": "Single",
              },
            ],
          },
          Object {
            "bathRooms": 4,
            "genderPolicy": "unisex",
            "houseName": "Guest House 3",
            "id": "bytrwqjk9",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Lagos, Nigeria",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "btrfdseq23",
                "roomName": "Aben",
                "roomType": "Ensuite",
              },
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "btrfdsnytr3",
                "roomName": "Aben",
                "roomType": "Ensuite",
              },
            ],
          },
          Object {
            "bathRooms": 1,
            "genderPolicy": "unisex",
            "houseName": "Guest House 4",
            "id": "requntm0",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Kigali, Rwanda",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "mnytbvfdc9",
                "roomName": "Berger",
                "roomType": "Suite",
              },
            ],
          },
        ]
      }
      handleOnRestore={[Function]}
      id="handleOnRestoreId"
      isLoading={false}
    />
  </div>
</Fragment>
`;
